// apps/music.js
var request = require('request');
var user_count = 0;
function Music(bot) {
  var vaggiMusicDB = [
    {
      name: 'Кристина Си',
      url: 'http://chavingston.com/telegrambot/music/kristina-si.mp3'
    },
    {
      name: 'Барак Обама',
      url: 'http://chavingston.com/telegrambot/music/obama.mp3'
    },
    {
      name: 'Ченнинг Татум',
      url: 'http://chavingston.com/telegrambot/music/chen.mp3'
    },
    {
      name: 'Бильбо Бэггинс',
      url: 'http://chavingston.com/telegrambot/music/bagins.mp3'
    }];
    var opts = {
      reply_markup: JSON.stringify({
        keyboard: [
          ['Кристина Си'],
          ['Барак Обама'],
          ['Ченнинг Татум'],
          ['Бильбо Бэггинс']
        ],
        one_time_keyboard: true
      })
    };
    bot.on('message', function (msg) {
      if (msg.text == '/start') {
        user_count += 1;
        console.log('Бот использовали: ' + user_count);
        console.log(msg.chat.first_name + " " + msg.chat.last_name + ' username: ' + msg.chat.username);
        bot.sendMessage(msg.chat.id, 'Салют камрады!\nДля Вас есть доступные команды:\n\/music - немного стиля для твоей толстушки.', {hide_keyboard: true});
      }
      if (msg.text == '/music') {
        console.log('ЗАКАЗЫЛИ МУЗЫКУ');
        bot.sendMessage(msg.chat.id, 'Выбери одну из моих песен, пока это еще бесплатно, хоуми:', opts);
      }
      else {
        for (var track in vaggiMusicDB) {
          if (msg.text == vaggiMusicDB[track].name) {
            var url = vaggiMusicDB[track].url;
            var audio = request(url);
            bot.sendMessage(msg.chat.id, 'Вы выбрали композицию: ' + vaggiMusicDB[track].name + '.'+'\nПодождите секундочку я ворую бит...');
            bot.sendAudio(msg.chat.id, audio);
          }
        }
      }
    });
}
exports.Music = Music;
