// apps/music.js
var request = require('request');
function Music(bot) {
  var vaggiMusicDB = [
    {
      name: 'Кристина Си',
      url: 'http://chavingston.com/telegrambot/music/chen.mp3'
    },
    {
      name: 'Барак Обама',
      url: 'http://chavingston.com/telegrambot/music/obama.mp3'
    },
    {
      name: 'Ченнинг Татум',
      url: 'http://chavingston.com/telegrambot/music/chen.mp3'
    },
    {
      name: 'Бильбо Бэггинс',
      url: 'http://chavingston.com/telegrambot/music/bagins.mp3'
    }];
    var opts = {
      reply_markup: JSON.stringify({
        keyboard: [
          ['Кристина Си'],
          ['Барак Обама'],
          ['Ченнинг Татум'],
          ['Бильбо Бэггинс']
        ],
        one_time_keyboard: true
      })
    };
    bot.on('message', function (msg) {
      if (msg.text == 'music') {
        bot.sendMessage(msg.chat.id, 'Выберите трек', opts);
      }
      else {
        for (var track in vaggiMusicDB) {
          if (msg.text == vaggiMusicDB[track].name) {
            var url = vaggiMusicDB[track].url;
            var audio = request(url);
            bot.sendMessage(msg.chat.id, 'Now playing: ' + vaggiMusicDB[track].name + 'by Monsieur Lapedi');
            bot.sendAudio(msg.chat.id, audio);
          }
        }
      }
    });
}
exports.Music = Music;
